Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1

```diff
@@ -560,3 +561,239 @@ function UpdateSettingsFile {
+            if ($originalSourceFolder -and ($file.origin -ne 'custom template')) {
+                Push-Location $sourceFolder
+                $relativePath = Resolve-Path -Path $srcFile -Relative # resolve the path relative to the current location (template folder)
+                Pop-Location
+                if (Test-Path (Join-Path $originalSourceFolder $relativePath) -PathType Leaf) {
+                    # If the file exists in the original template folder, use that file instead
+                    $fullFilePath.originalSourceFullPath = Join-Path $originalSourceFolder $relativePath -Resolve
+                }
+            }
+
+            $destinationName = $(Split-Path -Path $srcFile -Leaf)
+            if($file.Keys -contains 'destinationName' -and ($file.destinationName)) {
+                $destinationName = $file.destinationName
+            }
+
+            if($file.Keys -contains 'perProject' -and $file.perProject -eq $true) {
+                # Multiple file entries, one for each project
+                # Destination full path is the destination base folder + project + destinationPath + destinationName
+
+                foreach($project in $projects) {
+                    if($project -eq '.') {
+                        $project = '' # If project is '.', it means the root folder, so we use an empty string
+                    }
+
+                    $projectFile = $fullFilePath.Clone()
+
+                    $projectFile.destinationFullPath = Join-Path $destinationFolder $project
+                    $projectFile.destinationFullPath = Join-Path $projectFile.destinationFullPath $file.destinationPath
+                    $projectFile.destinationFullPath = Join-Path $projectFile.destinationFullPath $destinationName
+
+                    Write-Host "Adding per-project file for project '$project': sourceFullPath '$($projectFile.sourceFullPath)', originalSourceFullPath '$($projectFile.originalSourceFullPath)', destinationFullPath '$($projectFile.destinationFullPath)'"
+                    $fullFilePaths += $projectFile
+                }
+            }
+            else {
+                # Single file entry
+                # Destination full path is the destination base folder + destinationPath + destinationName
+
+                $fullFilePath.destinationFullPath = Join-Path $destinationFolder $file.destinationPath
+                $fullFilePath.destinationFullPath = Join-Path $fullFilePath.destinationFullPath $destinationName
+
+                Write-Host "Adding file: sourceFullPath '$($fullFilePath.sourceFullPath)', originalSourceFullPath '$($fullFilePath.originalSourceFullPath)', destinationFullPath '$($fullFilePath.destinationFullPath)'"
+                $fullFilePaths += $fullFilePath
+            }
+        }
+    }
+    # Remove duplicates
+    $fullFilePaths = $fullFilePaths | Sort-Object { $_.sourceFullPath}, { $_.originalSourceFullPath}, { $_.destinationFullPath}, { $_.type } -Unique

```

Comment:
Not totally sure I understand this.. Why are we sorting them based on those properties? Shouldn't we just get all the unique destinationFullPaths?